app/services/teacher_training_api/create_course.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+module TeacherTrainingApi
+  class CreateCourse
+    include ServicePattern
+
+    def initialize(provider:, course:)
+      @provider = provider
+      @attrs = course["attributes"]
+    end
+
+    def call
+      provider.courses.find_or_create_by!(name: attrs["name"])
```

Comment:
At the moment, we're just saving the name so there's no update for now. This is where the update on conflict logic will go.